src/more_ops.rs

```diff
@@ -17,59 +18,59 @@ use crate::serialize::node_to_bytes;
-const ARITH_COST_PER_LIMB_DIVIDER: u32 = 64;
+const ARITH_BASE_COST: Cost = 4;
+const ARITH_COST_PER_ARG: Cost = 8;
+const ARITH_COST_PER_LIMB_DIVIDER: Cost = 64;
 
-const LOG_BASE_COST: u32 = 6;
-const LOG_COST_PER_ARG: u32 = 8;
-const LOG_COST_PER_LIMB_DIVIDER: u32 = 64;
+const LOG_BASE_COST: Cost = 6;
+const LOG_COST_PER_ARG: Cost = 8;
+const LOG_COST_PER_LIMB_DIVIDER: Cost = 64;
 
-const LOGNOT_BASE_COST: u32 = 12;
-const LOGNOT_COST_PER_BYTE_DIVIDER: u32 = 512;
+const LOGNOT_BASE_COST: Cost = 12;
+const LOGNOT_COST_PER_BYTE_DIVIDER: Cost = 512;
 
-const MUL_BASE_COST: u32 = 2;
-const MUL_COST_PER_OP: u32 = 18;
-const MUL_LINEAR_COST_PER_BYTE_DIVIDER: u32 = 64;
-const MUL_SQUARE_COST_PER_BYTE_DIVIDER: u32 = 44500;
+const MUL_BASE_COST: Cost = 2;
+const MUL_COST_PER_OP: Cost = 18;
+const MUL_LINEAR_COST_PER_BYTE_DIVIDER: Cost = 64;
+const MUL_SQUARE_COST_PER_BYTE_DIVIDER: Cost = 44500;
 
-const GR_BASE_COST: u32 = 19;
-const GR_COST_PER_LIMB_DIVIDER: u32 = 64;
+const GR_BASE_COST: Cost = 19;
+const GR_COST_PER_LIMB_DIVIDER: Cost = 64;
 
-const CMP_BASE_COST: u32 = 16;
-const CMP_COST_PER_LIMB_DIVIDER: u32 = 64;
+const CMP_BASE_COST: Cost = 16;
+const CMP_COST_PER_LIMB_DIVIDER: Cost = 64;
 
-const STRLEN_BASE_COST: u32 = 18;
-const STRLEN_COST_PER_BYTE_DIVIDER: u32 = 4096;
+const STRLEN_BASE_COST: Cost = 18;
+const STRLEN_COST_PER_BYTE_DIVIDER: Cost = 4096;
 
-const CONCAT_BASE_COST: u32 = 4;
-const CONCAT_COST_PER_ARG: u32 = 8;
-const CONCAT_COST_PER_BYTE_DIVIDER: u32 = 830;
+const CONCAT_BASE_COST: Cost = 4;
+const CONCAT_COST_PER_ARG: Cost = 8;
+const CONCAT_COST_PER_BYTE_DIVIDER: Cost = 830;
 
-const DIVMOD_BASE_COST: u32 = 29;
-const DIVMOD_COST_PER_LIMB_DIVIDER: u32 = 64;
+const DIVMOD_BASE_COST: Cost = 29;
+const DIVMOD_COST_PER_LIMB_DIVIDER: Cost = 64;
 
-const DIV_BASE_COST: u32 = 29;
-const DIV_COST_PER_LIMB_DIVIDER: u32 = 64;
+const DIV_BASE_COST: Cost = 29;
+const DIV_COST_PER_LIMB_DIVIDER: Cost = 64;
 
-const SHA256_BASE_COST: u32 = 3;
-const SHA256_COST_PER_ARG: u32 = 8;
-const SHA256_COST_PER_BYTE_DIVIDER: u32 = 64;
+const SHA256_BASE_COST: Cost = 3;
+const SHA256_COST_PER_ARG: Cost = 8;
+const SHA256_COST_PER_BYTE_DIVIDER: Cost = 64;
 
-const SHIFT_BASE_COST: u32 = 21;
-const SHIFT_COST_PER_BYTE_DIVIDER: u32 = 256;
+const SHIFT_BASE_COST: Cost = 21;
+const SHIFT_COST_PER_BYTE_DIVIDER: Cost = 256;
 
-const BOOL_BASE_COST: u32 = 1;
-const BOOL_COST_PER_ARG: u32 = 8;
+const BOOL_BASE_COST: Cost = 1;
+const BOOL_COST_PER_ARG: Cost = 8;
 
-const POINT_ADD_BASE_COST: u32 = 213;
-const POINT_ADD_COST_PER_ARG: u32 = 358;
+const POINT_ADD_BASE_COST: Cost = 213;
+const POINT_ADD_COST_PER_ARG: Cost = 358;
 
-const PUBKEY_BASE_COST: u32 = 394;
-const PUBKEY_COST_PER_BYTE_DIVIDER: u32 = 4;
+const PUBKEY_BASE_COST: Cost = 394;
+const PUBKEY_COST_PER_BYTE_DIVIDER: Cost = 4;
 
-fn limbs_for_int(v: &Number) -> u32 {
-    ((v.bits() + 7) >> 3) as u32
+fn limbs_for_int(v: &Number) -> usize {
+    ((v.bits() + 7) / 8) as usize
```

Comment:
It's a matter of taste :)

Reply:
I saw you changed it in the other PR, so I figured I'd do it here too.. or did I misread that?

Reply:
No, but I did go to a rust compiler explorer to confirm that rust would compile them the same. It's exactly the same for `usize`, but very slightly different for `i32`. [Check it out](https://rust.godbolt.org/#g:!((g:!((g:!((h:codeEditor,i:(fontScale:14,j:1,lang:rust,selection:(endColumn:1,endLineNumber:18,positionColumn:1,positionLineNumber:18,selectionStartColumn:1,selectionStartLineNumber:1,startColumn:1,startLineNumber:1),source:'pub+fn+div1(num:+usize)+-%3E+usize+%7B%0A++++num+/+8%0A%7D%0A%0Apub+fn+div2(num:+usize)+-%3E+usize+%7B%0A++++//+this+one+doesn!'t+even+appear+as+it!'s+just+aliased+to+div1%0A++++num+%3E%3E+3%0A%7D%0A%0Apub+fn+div3(num:+i32)+-%3E+i32+%7B%0A++++num+/+8%0A%7D%0A%0Apub+fn+div4(num:+i32)+-%3E+i32+%7B%0A++++num+%3E%3E+3%0A%7D%0A%0A'),l:'5',n:'0',o:'Rust+source+%231',t:'0')),k:50,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((h:compiler,i:(compiler:r1500,filters:(b:'0',binary:'1',commentOnly:'0',demangle:'0',directives:'0',execute:'1',intel:'0',libraryCode:'0',trim:'1'),fontScale:14,j:1,lang:rust,libs:!(),options:'-O',selection:(endColumn:1,endLineNumber:1,positionColumn:1,positionLineNumber:1,selectionStartColumn:1,selectionStartLineNumber:1,startColumn:1,startLineNumber:1),source:1),l:'5',n:'0',o:'rustc+1.50.0+(Editor+%231,+Compiler+%231)+Rust',t:'0')),k:50,l:'4',n:'0',o:'',s:0,t:'0')),l:'2',n:'0',o:'',t:'0')),version:4)